### PR TITLE
Use FF abuse report panel for extensions and static themes only

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-frontend/issues/8929

---

This would be a good candidate for cherry-picking.

### CODE REVIEW

File: src/amo/components/ReportAbuseButton/index.js [hunk 1]
```diff
@@ -81,13 +82,26 @@ export class ReportAbuseButtonBase extends React.Component<InternalProps> {
...
   onReportButtonClick = async (event: SyntheticEvent<any>) => {
+    const { _hasAbuseReportPanelEnabled, addon, dispatch } = this.props;
+
     event.preventDefault();
 
-    const { _hasAbuseReportPanelEnabled, addon, dispatch } = this.props;
-    if (_hasAbuseReportPanelEnabled()) {
+    if (
+      _hasAbuseReportPanelEnabled() &&
+      // The integrated abuse report panel currently support only extension and
```
Comment (bobsilverberg) [inline src/amo/components/ReportAbuseButton/index.js]: Nit: s/support/supports  & s/extension/extensions

File: src/amo/components/ReportAbuseButton/index.js [hunk 2]
```diff
@@ -81,13 +82,26 @@ export class ReportAbuseButtonBase extends React.Component<InternalProps> {
...
+      // themes (e.g. we do only have abuse categories and related localized
+      // descriptions only for these two kind of addons), and so currently it
+      // is going to refuse to create an abuse report panel for langpacks,
+      // dictionaries and search tools.
+      //
+      // Static themes should be supported but there is a bug in FF, see:
+      // https://github.com/mozilla/addons-frontend/issues/8762#issuecomment-553430081
```
Comment (bobsilverberg) [inline src/amo/components/ReportAbuseButton/index.js]: We should file a follow-up to add static themes to this list once the FF bug is fixed.

File: src/amo/components/ReportAbuseButton/index.js [hunk 3]
```diff
@@ -81,13 +82,26 @@ export class ReportAbuseButtonBase extends React.Component<InternalProps> {
...
+      [ADDON_TYPE_EXTENSION].includes(addon.type)
```
Comment (bobsilverberg) [inline src/amo/components/ReportAbuseButton/index.js]: Nit: we don't really need `includes` for a single item, but it positions us for the follow-up.

File: tests/unit/amo/components/TestReportAbuseButton.js [hunk 4]
```diff
@@ -135,26 +143,54 @@ describe(__filename, () => {
...
-    root.find(Button).simulate('click', fakeEvent);
-
-    sinon.assert.called(fakeEvent.preventDefault);
-    sinon.assert.calledWith(
-      dispatchSpy,
-      initiateAddonAbuseReportViaFirefox({ addon }),
-    );
-  });
+  it.each([ADDON_TYPE_EXTENSION])(
+    'initiates an abuse report via Firefox when the "report" button is clicked if supported and add-on is %s',
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestReportAbuseButton.js]: Nit: s/add-on/add-on type

File: tests/unit/amo/components/TestReportAbuseButton.js [hunk 5]
```diff
@@ -135,26 +143,54 @@ describe(__filename, () => {
...
+    ADDON_TYPE_DICT,
+    ADDON_TYPE_LANG,
+    ADDON_TYPE_OPENSEARCH,
+    ADDON_TYPE_STATIC_THEME,
+  ])(
+    'does not initiate an abuse report via Firefox when add-on is %s',
+    (addonType) => {
+      const _hasAbuseReportPanelEnabled = sinon.stub().returns(true);
+      const addon = createInternalAddon({ ...fakeAddon, type: addonType });
+      const fakeEvent = createFakeEvent();
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestReportAbuseButton.js]: `createFakeEvent()` could be inlined below as we're not asserting on it.

File: src/amo/components/ReportAbuseButton/index.js (hunk 2 above)
Comment (willdurand) [reply src/amo/components/ReportAbuseButton/index.js]: Good call, thanks. Done here: https://github.com/mozilla/addons-frontend/issues/8931

**Pull request merged**